actionview/test/template/url_helper_test.rb

```diff
@@ -15,6 +15,10 @@ class UrlHelperTest < ActiveSupport::TestCase
     get "/other" => "foo#other"
     get "/article/:id" => "foo#article", :as => :article
     get "/category/:category" => "foo#category"
+
+    scope :engine do
+      get "/" => "foo#bar"
```

Comment:
Could you explain how this is testing what happens in a real engine and why we need to clear `request_uri` to make it work?